### PR TITLE
Docs: Document Makefile

### PR DESCRIPTION
**Pull request by marcusolsson**
**What this PR does / why we need it**:

This pull request documents the Makefile and adds a `help` target for displaying help texts.

Here's an example of the information displayed when running `make`, or `make help`.

makefile

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -100,33 +107,39 @@ gosec: scripts/go/bin/gosec
...
+##@ Development environment
 
 # create docker-compose file with provided sources and start them
 # example: make devenv sources=postgres,openldap
 ifeq ($(sources),)
 devenv:
 	@printf 'You have to define sources for this command \nexample: make devenv sources=postgres,openldap\n'
 else
-devenv: devenv-down
+devenv: devenv-down ## Set up development environment.
```
Comment (torkelo) [inline Makefile]: > Set up development environment

This seems like something you need to do in order to develop. 
Maybe something like this:

Start optional external dependencies via docker-compose like postgres, mysql, prometheus, openldap, elasticsearch etc.

Or

Start optional services of postgres, mysql, prometheus, openldap, elasticsearch, etc (you choose)

**Review by aknuds1 — CHANGES_REQUESTED** [review]
I think it mainly looks useful, but I question whether it's a good idea to drop the default `make` target. @marefr What do you think?

**Comment by marcusolsson** [comment]
Thanks for the feedback!

**Review by marefr — APPROVED** [review]
LGTM

**Review by aknuds1 — APPROVED** [review]
LGTM

**Pull request merged**